.ci/ci-fast-return.sh

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description: Perform CI 'fast path return' checks.
+# Returns: 1 for 'fast return OK', 0 for 'do not fastpath'.
+#
+# Checks for:
+#  - file name patterns from the YAML that will force the CI to run
+#  - file name patterns from the YAML to see if we can skip all files in the PR
+#  - If potentially skipping, looks for a 'force' label on the PR to force the CI
+#    to run anyway
+
+set -e
+
+[ -n "$DEBUG" ] && set -x
+
+script_name="${0##*/}"
+script_gitname=".ci/${script_name}"
```

Comment:
Hard-coded path. I think you could do something like this for all the `*_gitname` vars that should continue to work even if this file gets moved to a different sub-dir:

```sh
script_gitname=$(git ls-files --full-name ${script_name}")
```

Or this would be safer:

```sh
script_gitname=$(git ls-files --full-name | grep "/${script_name}$")
```

Reply:
To do that I think we have to ensure we are inside the repo where the file lives (it does not work if you are not in the git tree of the repo you are examining afaict), and we need to full path inside the repo to where the file we are looking for is (if you just try `ci-fast-path.sh` at the top level of the repo, you get no name). So, I think it is a bit catch22 - you have to know what path you are looking in in order to get its, err, path.
Thus, I think the hard wiring of the '.ci' subdir has to happen soemwhere, and doing it where it is right now I think is as good as anywhere. Unless you can show me some code that is going to work no matter where/how the script is invoked?

Reply:
That's a good point and it may be worth adding a comment / note in the commit explaining where we expect this script to be run from. Well, if we can't use `basename $cidir`, how about a top-level variable set to `.ci/` along with a juicy variable? That atleast confines the issue to a central location, rather than hard-coding `.ci/` in the three `*_gitname` variables.

Reply:
basedir in a var I can do ... done